lib/internal/streams/operators.js

```diff
@@ -150,6 +151,40 @@ async function * map(fn, options) {
   }
 }
 
+async function some(fn, options) {
+  // https://tc39.es/proposal-iterator-helpers/#sec-iteratorprototype.some
+  // Note that some does short circuit but also closes the iterator if it does
+  const ac = new AbortController();
+  if (options?.signal) {
+    if (options.signal.aborted) {
+      ac.abort();
+    }
+    options.signal.addEventListener('abort', () => ac.abort(), {
+      [kWeakHandler]: this,
+      once: true,
+    });
+  }
+  const mapped = this.map(fn, { ...options, signal: ac.signal });
+  for await (const result of mapped) {
+    if (result) {
+      ac.abort();
+      return true;
+    }
+  }
+  return false;
+}
```

Comment:
On a second thoughts, I don't think it makes much sense to have this function in Node.js. I don't see how useful it can be for people.

Reply:
I've thought of a few examples where I'd use this, mostly with object mode streams. To be fair it's here mostly for parity with the proposal but users do gain concurrency and cancellation configuration over `AsyncIterator.from(...).some(...)` over the stream.

For example:

```js
const anyAvailable = await Readable.from(apiList).some(async (api, { signal }) => {
  const { statusCode } = await request(api, { signal });
  return statusCode === 200; 
}, { concurrency: 8, signal }); // signal from the outside
```

Basically any case where I want to ask an "is any" (or "is all") question about stuff asynchronously but also need to limit concurrency and cancellation which are features "over" what I'd get with a regular async iterator and not a stream.

Reply:
Ok, let's do this.